src/python/pants/rules/core/test.py

```diff
@@ -142,7 +159,18 @@ class AddressAndDebugRequest:
   results = await MultiGet(Get[AddressAndTestResult](Address, addr.to_address()) for addr in addresses)
   did_any_fail = False
   filtered_results = [(x.address, x.test_result) for x in results if x.test_result is not None]
-
+  if options.values.run_coverage:
+    # TODO(#8915) Make a generic interface for collecting coverage reports. This class shouldn't have to know about
+    # PytestCoverageReport and JunitCoverageReport etc. It should only know about CoverageReport.
+    pytest_coverage_report = await Get[PytestCoverageReport](
+      AddressAndTestResults,
+      AddressAndTestResults(tuple(sorted(filtered_results)))
```

Comment:
No need to `sort`. The order is already stable because line 161 is a list comprehension. Also, you can change line 161 to be `tuple(x for x in xs)` so that you don't need to cast this to a tuple here and can simply pass `filtered_results`.